src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionPath.java

```diff
@@ -0,0 +1,152 @@
+package io.stargate.sgv2.jsonapi.service.projection;
+
+import io.stargate.sgv2.jsonapi.exception.ProjectionException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An immutable representation of a projection path composed of segments.
+ *
+ * <p>A projection path is created from an encoded string (e.g., "pricing.price&amp;.usd"), where
+ * segments are separated by dots ('.') and the escape character ('&amp;') allows literal dots or
+ * ampersands within segments.
+ *
+ * <p>Use {@link #from(String)} to create an instance, {@link #getSegments()} to access the
+ * segments, and {@link #encode()} to get the dot-separated path string.
+ */
+public final class ProjectionPath {
+  private final List<String> segments;
+
+  private ProjectionPath(List<String> segments) {
+    // Make a defensive copy to preserve immutability.
+    this.segments = Collections.unmodifiableList(new ArrayList<>(segments));
+  }
+
+  /**
+   * Factory method to create a ProjectionPath from the given encoded path string. Handles escape
+   * sequences where '&' escapes a dot or an ampersand.
+   *
+   * @param path the encoded path string (e.g., "pricing.price&.usd")
+   * @return a new ProjectionPath instance containing the decoded segments
+   * @throws IllegalArgumentException if the escape sequence is invalid
+   */
+  public static ProjectionPath from(String path) {
+    Objects.requireNonNull(path, "path cannot be null");
+    return new ProjectionPath(decode(path));
+  }
+
+  /**
+   * Returns the list of decoded segments.
+   *
+   * @return an unmodifiable list of segments
+   */
+  public List<String> getSegments() {
```

Comment:
As per earlier notes, replace with two methods:

* `public int segment(ind index)`
* `public int segmentCount()` (or, if you prefer, `int size()`